plugins/chat/lib/chat/message_mover.rb

```diff
@@ -186,10 +186,11 @@ def delete_source_messages
     end
 
     def add_moved_placeholder(destination_channel, first_moved_message)
-      Chat::MessageCreator.create(
-        chat_channel: @source_channel,
-        user: Discourse.system_user,
-        content:
+      @source_channel.add(Discourse.system_user)
```

Comment:
The system user needs to have access to the channel like any other regular user. I think this is something we can see with bots on other platforms (slack, etc.), if you don’t give them permissions to write to certain channels, they won’t be able to do it.
It’s also simpler to just apply the same logic to any user, instead of having to hack something for special users.